nuget/lib/dependabot/nuget/nuget_config_credential_helpers.rb

```diff
@@ -68,6 +68,9 @@ def self.patch_nuget_config_for_action(credentials, &_block)
         add_credentials_to_nuget_config(credentials)
         begin
           yield
+        rescue DependabotError
+          # forward these
+          raise
```

Comment:
Does I assume this Only does the more specific rescue and not both?

Reply:
Correct, this is a very specific rescue because the `yield` statement on line 70 is what calls the C# CLI and the response of that could get turned into a `PrivateSourceAuthenticationFailure` which inherits from `DependabotError` so all of those errors need to be reported up the chain.